packages/tizen_rpc_port/example/example_proxy/lib/message_proxy.dart

```diff
@@ -0,0 +1,262 @@
+    _once = parcel.readBool();
+  }
+}
+
+/// The NotifyCB class to invoke the delegate method.
+abstract class NotifyCB extends CallbackBase {
+  /// Constructor for this class.
+  NotifyCB({bool once = false}) : super(_DelegateId.notifyCB.id, once);
+
+  /// This abstract method will be called when the delegate is received event.
+  Future<void> onReceived(String sender, String msg);
+
+  @override
+  Future<void> _onReceivedEvent(Parcel parcel) async {
+    final String sender = parcel.readString();
+    final String msg = parcel.readString();
+
+    await onReceived(sender, msg);
+  }
+}
+
+/// Abstract class for creating [Message] class for RPC.
+abstract class Message extends ProxyBase {
+  /// Constructor for this class.
+  Message(String appid) : super(appid, 'Message');
+
+  bool _online = false;
+  final List<CallbackBase> _delegateList = <CallbackBase>[];
+
+  /// This abstract method will be called when the connet() is succeed.
+  Future<void> onConnected();
+
+  /// This abstract method will be called when the connection with the stub is disconnected.
+  Future<void> onDisconnected();
+
+  /// This abstract method will be called when connect() is failed.
+  Future<void> onRejected(String errorMessage);
+
+  @override
+  @nonVirtual
+  Future<void> onConnectedEvent() async {
+    _online = true;
+    await onConnected();
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onDisconnectedEvent() async {
+    _online = false;
+    await onDisconnected();
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onRejectedEvent(String errorMessage) async {
+    await onRejected(errorMessage);
+  }
+
+  @override
+  @nonVirtual
+  Future<void> onReceivedEvent(Parcel parcel) async {
+    final int cmd = parcel.readInt32();
+    if (cmd != _MethodId.callback.id) {
+      return;
+    }
+
+    await _processReceivedEvent(parcel);
+  }
+
+  Future<void> _processReceivedEvent(Parcel parcel) async {
+    final int id = parcel.readInt32();
+    final int seqId = parcel.readInt32();
+    final bool once = parcel.readBool();
+
+    for (final CallbackBase delegate in _delegateList) {
+      if (delegate._id == id && delegate._seqId == seqId) {
+        await delegate._onReceivedEvent(parcel);
+        if (delegate._once && once) {
+          _delegateList.remove(delegate);
+        }
+        break;
+      }
+    }
+  }
+
+  Future<Parcel> _consumeCommand(Port port) async {
+    while (true) {
+      try {
+        final Parcel parcel = Parcel.fromPort(port);
+        final int cmd = parcel.readInt32();
+        if (cmd == _MethodId.result.id) {
+          return parcel;
+        }
+      } catch (e) {
+        Log.error(_logTag, e.toString());
+        return Parcel();
+      }
+    }
+  }
+
+  /// Connects with the stub application.
+  @override
+  Future<void> connect() async {
+    Log.info(_logTag, 'connect()');
+    await super.connect();
+  }
+
+  /// Dispose registered delegate interface.
+  void disposeCallback(String tag) {
```

Comment:
Who calls this method and when should it be called? (maybe in `unregister`?)

Reply:
It is not currently used in the sample app.
However, when 'unregister' is called in proxy, the callback is disabled and the message is not received after calling send().

Reply:
Wait..what? Are you sure you're talking about `disposeCallback`?

Reply:
`disposeCallback` is default method provided from TIDL.
`unregister` method is user defined method. So it depends on the user's implementation.
In our sample app case, disposing callbacks has been implemented in `unregister`.
But is is just example.

Reply:
I can't understand even a single line of your comment. `unregister` is not being used in the app code either. There's no such code that "disposes" the callbacks.

Reply:
I added both `register` & `unregister` function buttons.
you can test unregister `NotifyCallback` as these buttons.

Reply:
Hmm. The question still has not been answered. When should `disposeCallback` be called?